### DISCUSSION TITLE
Unexpected spliced/unspliced percentages

### DISCUSSION THREAD

**Discussion by carbycrab** [theislab/scvelo#899] [Q&A]
Hello,

Thank you for creating such an innovative, streamlined package! I am using `scvelo` in my single cell analyses and had a question regarding the scvelo results I got. When I checked the spliced/unspliced percentages `scv.pl.proportions(adata)`, I got majority unspliced (86%) and minority spliced (14%), which seem way off from the reported 15-25% of unspliced intronic reads in La Manno et al 2018. I was wondering whether these statistics indicate something wrong with my analysis method or I should take this as factual and trust my downstream results. Here's my method: I used `velocyto` to generate individual loom files for all 9 of my 10x Chromium samples, merged the data, and subset to select 18555 genes x 14813 cells of interest. The downstream analyses presented in `scvelo` tutorial page all ran successfully. 

Thanks so much for the help!

-Angela

**Comment by WeilerP** [comment]
@carbycrab, seems like something is off but without further info/access to the data I can't offer much insight. Are you checking the proportions on the raw counts? If not, this can have unexpected behavior. See e.g. #216 or #462.

**Reply by carbycrab** [reply]
Hi @WeilerP,

Thanks so much for the quick response and issue references. 

I believe I did check the proportions on the raw counts. I ran the proportions before `scv.pp.filter_and_normalize` if this is what you mean by checking proportions on raw counts. I only filtered the cells and genes, based on our seurat analysis, but the spliced/unspliced counts are from `velocyto 10x` pipeline which uses cellranger's `outs/filtered_feature_matrices` as input. In any case, I also checked the splice/unspliced percentages without subsetting any genes or cells and still got the same percentages as above.

This is what I got when I ran  (#216)
```
counts_s = scv.utils.sum_var(adata.layers['spliced'])
counts_u = scv.utils.sum_var(adata.layers['unspliced'])
fractions_u = counts_u / (counts_s + counts_u)
scv.pl.scatter(adata, color=fractions_u, basis= 'umap')
```
(I got an attribute error when specifying argument `smooth= True`)
<img width="409" alt="image" src="https://user-images.githubusercontent.com/55811746/178595900-66ac9934-b26d-478f-a20e-ff9550959fcc.png">

When I tried to run `scv.core.get_initial_size(adata)` in #575, I got:
`Out: array([1393., 2806., 1830., ...,  986., 1095.,  363.], dtype=float32)`
but I don't see the columns `initial_size_unspliced` and `initial_size_spliced` in `adata.obs`

Here are the versions I am using
`scvelo==0.2.4  scanpy==1.7.2  anndata==0.7.6  loompy==3.0.6  numpy==1.21.5  scipy==1.7.3  matplotlib==3.3.4  sklearn==1.0.2  pandas==1.2.3   `

**Reply by WeilerP** [reply]
@carbycrab, okay. With raw counts I mean the unprocessed, whole counts.
Two things: `scv.core.get_initial_size(adata)` only returns the size (there is no saving going on, you'll need `set_initial_size` for that) and does so for `adata.X`. To get the initial size of the unspliced and spliced layer, you need to call `scv.core.get_initial_size(adata, layer='unspliced')` (`layer=spliced`, respectively). Could you please double-check that the plotted proportions are the same as reported? So something along the lines of

```python
initial_size_unspliced = scv.core.get_initial_size(adata, layer='unspliced')
initial_size_spliced = scv.core.get_initial_size(adata, layer='spliced')

total_counts = np.sum([initial_size_unspliced, initial_size_spliced], axis=0)

# Might be worth checking the distribution of this
fraction_unspliced = initial_size_unspliced / total_counts
mean_fraction_unspliced = fraction_unspliced.mean() # This is what's being plotted

fraction_spliced = initial_size_spliced / total_counts
mean_fraction_spliced = fraction_spliced.mean()
```

**Reply by carbycrab** [reply]
I see. Thank you so much for the code. I have `initial_size_unspliced` and `initial_size_spliced` in `adata.obs` now. When I ran your code, I got:
```
total_counts = [ 9777 14572 11387 ...  7308  7287  2530]
fraction_unspliced = [0.85752276 0.80743892 0.83929042 ... 0.86507937 0.8497324  0.85652174]
mean_fraction_unspliced = 0.8555956615794414
fraction_spliced = [0.85752276 0.80743892 0.83929042 ... 0.86507937 0.8497324  0.85652174]
mean_fraction_spliced = 0.8555956615794414
```

Do you know why the unspliced and spliced percentages are exactly the same? The unspliced percentage matches the pie chart, but the spliced reflects 86% instead of 14%.

**Reply by WeilerP** [reply]
@carbycrab I suppose you copied my type on the code for the fraction of spliced counts. I updated it accordingly.

**Reply by carbycrab** [reply]
Oh thanks. Sorry, I should've checked that more carefully. Yes after running it again, the fractions match the pie chart-- 86% unspliced, 14% spliced. So, any other ideas for why this could be happening? Thanks so much.

**Reply by carbycrab** [reply]
I also tried this same method with another 10x dataset I have and obtained more normal values for spliced (82%) and unspliced (18%). This current dataset with strange results is also cell hashed. Could this be contributing to the abnormal proportions?

**Comment by Shabbirm21** [comment]
Hi @carbycrab and @WeilerP, I realize this a quite old post but I was wondering ifany of  you managed to figure out why you were getting this unusual behaviour? I am also getting 60% unspliced, 35% spliced and 5% ambguos. Any clue if this is a normal behavior? 


Thanks,
Madiha
